services/user/Chainmail/service/src/tables.rs

```diff
@@ -0,0 +1,20 @@
+use psibase::{AccountNumber, Fracpack, TableRecord};
+use serde::{Deserialize, Serialize};
+
+impl TableRecord for SavedMessage {
+    type PrimaryKey = u64;
+
+    const SECONDARY_KEYS: u8 = 0;
+
+    fn get_primary_key(&self) -> Self::PrimaryKey {
+        self.event_id
+    }
+}
```

Comment:
Is this because you can't use #[primary_key] when the struct is split into a separate module?

Reply:
Yes, I'm pretty sure #[primary_key] is macro-replaced by this impl. This is just what it looks like when the macro can't do it for you